packages/flutter/lib/src/material/expansion_panel.dart

```diff
@@ -400,7 +400,9 @@ class _ExpansionPanelListState extends State<ExpansionPanelList> {
             padding: _kExpandIconPadding,
             splashColor: child.splashColor,
             highlightColor: child.highlightColor,
-            onPressed: (bool isExpanded) => _handlePressed(isExpanded, index),
+            onPressed: !child.canTapOnHeader
```

Comment:
@nate-thegrate How come we are back to using disabled color and disabling the button?

Reply:
The change unfortunately was causing a few different tests to fail: https://github.com/flutter/flutter/runs/32046612820

I wanted to see if reverting would fix it 😄

Reply:
@nate-thegrate I will looking into a fix for the test that addresses the semantic issue but keeps the original fix. This change is kinda going back to what was causing the issue, just not really showing the issue because of the behaviour of IgnorePointer.

Reply:
Fair point.

I can revert the recent changes, and from there you could figure out how to fix the failing checks.
Would you like me to do that?